com.avaloq.tools.ddk.xtext.ui.test/src/com/avaloq/tools/ddk/xtext/ui/templates/TemplateProposalProviderHelperTest.xtend

```diff
@@ -0,0 +1,259 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+
+package com.avaloq.tools.ddk.xtext.ui.templates
+
+import com.google.inject.Guice
+import org.eclipse.jface.text.IDocument
+import org.eclipse.jface.text.IRegion
+import org.eclipse.jface.text.Position
+import org.eclipse.jface.text.templates.Template
+import org.eclipse.xtext.XtextRuntimeModule
+import org.eclipse.xtext.ui.editor.templates.XtextTemplateContext
+import org.eclipse.xtext.ui.editor.templates.XtextTemplateContextType
+import org.junit.AfterClass
+import org.junit.BeforeClass
+import org.junit.Test
+
+import static org.junit.Assert.assertArrayEquals
+import static org.junit.Assert.assertEquals
+import static org.mockito.Matchers.anyInt
+import static org.mockito.Mockito.mock
+import static org.mockito.Mockito.spy
+import static org.mockito.Mockito.verify
+import static org.mockito.Mockito.when
+
+public class TemplateProposalProviderHelperTest {

```

Comment:
i'd rather use java here

Reply:
I tried, and I'd rather not! '''"''' and '''\"''' are far more readable and less error-prone than their Java equivalents, "\"" and "\\\"".